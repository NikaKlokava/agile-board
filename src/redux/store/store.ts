import { legacy_createStore } from "redux";
import { boardsReducer } from "../reducers/boardReducer";

const store = legacy_createStore(boardsReducer);

export default store;

// export type RootState = ReturnType<typeof store.getState>

// export interface RootState {
//     user: UserState;
//     loader: LoaderState;
//   }
// export type AppDispatch = typeof store.dispatch
