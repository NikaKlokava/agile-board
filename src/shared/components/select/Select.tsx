import { Field } from "formik";
import { memo } from "react";
import { useAppSelector } from "../../../redux/hooks/hook";
import { FieldWrapper } from "../field_wrapper";

export const Select = memo(() => {
  const activeBoard = useAppSelector((state) => state.activeBoard);

  return (
    <FieldWrapper fieldName={"Current Status"}>
      <Field as="select" name="columnTitle">
        {activeBoard.columns.map((column, i) => (
          <option key={i} value={column.title}>
            {column.title}
          </option>
        ))}
      </Field>
    </FieldWrapper>
  );
});
