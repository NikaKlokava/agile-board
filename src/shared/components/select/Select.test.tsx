import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import store from "../../../redux/store/store";
import { Select } from "./Select";

jest.mock("./Select");

describe("Test the Select component", () => {
  test("The Select renders correctly", () => {
    const selectSnap = renderer
      .create(
        <Provider store={store}>
          <Select />
        </Provider>
      )
      .toJSON();
    expect(selectSnap).toMatchSnapshot();
  });
});
