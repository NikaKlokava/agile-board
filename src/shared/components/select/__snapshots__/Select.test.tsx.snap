// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Input component The Input renders correctly 1`] = `
<select>
  <option>
    Todo
  </option>
  <option>
    Doing
  </option>
  <option>
    Done
  </option>
</select>
`;

exports[`Test the Select component The Select renders correctly 1`] = `
<select>
  <option>
    Todo
  </option>
  <option>
    Doing
  </option>
  <option>
    Done
  </option>
</select>
`;
